### PR TITLE
Merge branch csp-metadata to main

### PR DESCRIPTION
**Pull request by KeithMnemonic**
This PR does a simple cherry pick of the changes merged to csp-metadata to main.

### CODE REVIEW

File: tests/unit/test_plugin.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    'tests/data/good_config.yaml',
+    pm.hook
+)
+
+
+def test_setup():
+    plugin.setup_adapter(config)  # Currently no-op
+
+
+def test_meter_billing(mock_boto3):
```
Comment (rtamalin) [inline tests/unit/test_plugin.py:38]: Should probably have a `pass` or docstring for this and the next test...

Comment (rtamalin) [reply tests/unit/test_plugin.py:38]:
```suggestion
def test_meter_billing(mock_boto3):
    pass
```

File: tests/unit/test_plugin.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def test_meter_billing_error(mock_boto3):
```
Comment (rtamalin) [inline tests/unit/test_plugin.py:41]:
```suggestion
def test_meter_billing_error(mock_boto3):
    pass
```

**Comment by KeithMnemonic** [comment]
redoing correctly

**Pull request closed without merging**